consensus/istanbul/core/vrank.go

```diff
@@ -0,0 +1,240 @@
+// Copyright 2023 The klaytn Authors
+// This file is part of the klaytn library.
+//
+// The klaytn library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// The klaytn library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the klaytn library. If not, see <http://www.gnu.org/licenses/>.
+
+package core
+
+import (
+	"encoding/hex"
+	"fmt"
+	"math/big"
+	"sort"
+	"time"
+
+	"github.com/klaytn/klaytn/common"
+	"github.com/klaytn/klaytn/consensus/istanbul"
+	"github.com/rcrowley/go-metrics"
+)
+
+type Vrank struct {
+	startTime            time.Time
+	view                 istanbul.View
+	committee            istanbul.Validators
+	threshold            time.Duration
+	commitArrivalTimeMap map[common.Address]time.Duration
+}
+
+var (
+	// VRank metrics
+	vrankFirstCommitArrivalTimeGauge           = metrics.NewRegisteredGauge("vrank/first_commit", nil)
+	vrankQuorumCommitArrivalTimeGauge          = metrics.NewRegisteredGauge("vrank/quorum_commit", nil)
+	vrankAvgCommitArrivalTimeWithinQuorumGauge = metrics.NewRegisteredGauge("vrank/avg_commit_within_quorum", nil)
+	vrankLastCommitArrivalTimeGauge            = metrics.NewRegisteredGauge("vrank/last_commit", nil)
+
+	vrankDefaultThreshold = "300ms"
```

Comment:
It would be more helpful if you add a comment about what vrankDefaultThreshold is.

Reply:
added in e4036a59

Reply:
Thank you.